platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/lifecycle/AppExitLogger.kt

```diff
@@ -60,29 +76,40 @@ internal class AppExitLogger(
         logPreviousExitReasonIfAny()
     }
 
+    @WorkerThread
     fun uninstallAppExitLogger() {
+        checkIsNotOnMainThread()
         crashHandler.uninstall()
     }
 
     @TargetApi(Build.VERSION_CODES.R)
+    @WorkerThread
     fun saveCurrentSessionId(sessionId: String? = null) {
-        if (!runtime.isEnabled(RuntimeFeature.APP_EXIT_EVENTS) || !versionChecker.isAtLeast(Build.VERSION_CODES.R)) {
-            return
-        }
+        runOnBackgroundThread {
```

Comment:
I think it's overkill to move this operation to the bg, moreso this could also lead to some unpredictable race conditions (the sessionId might be be persisted on time to be read after)